*: drop dialog

arithx opened the pull request:
Switches to `echo "" >&2` and updates the `StandardOutput` of the
`coreos-installer.service` to `kmsg+console`. This makes the installer
be non-interactive. Later patches will drop into a shell in the
initramfs if all required parameters are not given.

dustymabe commented:
On a successful install I see:

```
[  OK  ] Reached target CoreOS Installer Target.
[  OK  ] Started CoreOS Installer.                               
[  OK  ] Reached target Remote File Systems (Pre).   
[    7.522620] coreos-installer[683]:   PID TTY          TIME CMD          
[    7.524060] coreos-installer[683]:   698 tty2     00:00:00 curl
[    8.546166] coreos-installer[683]:   PID TTY          TIME CMD
[    8.547657] coreos-installer[683]:   698 tty2     00:00:00 curl
[    8.589230] coreos-installer[683]: 24
[    9.620445] coreos-installer[683]:   PID TTY          TIME CMD 
[    9.621858] coreos-installer[683]:   698 tty2     00:00:00 curl          
[    9.658000] coreos-installer[683]: 52         
[   10.694455] coreos-installer[683]:   PID TTY          TIME CMD
[   10.696422] coreos-installer[683]:   698 tty2     00:00:01 curl
[   10.743443] coreos-installer[683]: 83                                  
[   11.764424] coreos-installer[683]:   PID TTY          TIME CMD
[   11.767527] coreos-installer[683]: Wiping /dev/sda
[   39.659360] coreos-installer[683]: embedding provided networking options
[   39.706754] coreos-installer[683]: Install complete
[   44.727146] coreos-installer[683]: umount:
         Stopping CoreOS Installer...
```

- The ps output is confusing - maybe we silence that

arithx commented on dracut/30coreos-installer/coreos-installer.service:
```diff
@@ -7,6 +7,7 @@ Type=simple
 ExecStart=/usr/libexec/coreos-installer
 #StandardInput=tty-force
 StandardInput=tty
+StandardOutput=journal+console
```
It was lost in a rebase and I just picked `journal+console`, I'll update back to `kmsg`

arithx commented on coreos-installer:
```diff
@@ -620,10 +616,46 @@ main() {
     fi
 }
 
+USAGE="Usage: $0 [options]
```
I initially had it in a function and it seemed to be breaking `getopts`

dustymabe replied:
Did you try calling the function with `$@` as an argument. i.e. `process_args $@`

arithx commented:
Hmm, my local builds are hitting the emergency shell, I'll go back through and make sure I don't have something out of date that's causing a discrepancy.

arithx commented:
Updated.

Looks like the `OnFailure` was moved from `coreos-installer.service` to `coreos-installer.target` in #23 but the service runs `After=` to the target so it didn't trigger through. I've re-added it to `coreos-installer.service`

----

Other notable changes in this last patch:

- Switched output lines to use a `log` function which both writes to `/tmp/debug` as well as writing to `stderr`
- Added `-f` to the Ignition curl, which as a by-product will now properly error when receiving 404's.
- Moved argument parsing into a `parse_args` function.
- Switched back to `kmsg+console` for `StandardOutput` in `coreos-installer.service`

dustymabe commented:
Emergency.target seems to work now and the output from a successful install looks better:

```
coreos-installer[683]: Image size is 666989080                             
coreos-installer[683]: tmpfs sized to 686 MB                               
coreos-installer[683]: IGNITION_URL IS http://192.168.1.101:8000/config.ign
coreos-installer[683]: Selected device is /dev/sda                         
coreos-installer[683]: Mounting tmpfs                                      
coreos-installer[683]: Downloading install image                           
coreos-installer[683]: 25%                                                 
coreos-installer[683]: 52%                                                 
coreos-installer[683]: 79%                                                 
coreos-installer[683]: Wiping /dev/sda
coreos-installer[683]: Writing disk image
coreos-installer[683]: embedding provided networking options
coreos-installer[683]: Install complete
coreos-installer[683]: umount: /mnt/boot_partition: not mounted.
```

One nit might be to change `embedding` to `Embedding` to match most of the other log statements capitalization.

arithx commented on coreos-installer:
```diff
@@ -497,36 +505,34 @@ mount_tmpfs() {
         PART_FILE_SIZE=$(ls -l /mnt/dl/imagefile.gz | awk '{print $5}') 2>/dev/null
         PCT=$(dc -e"2 k $PART_FILE_SIZE $IMAGE_SIZE / 100 * p" | sed -e"s/\..*$//" 2>/dev/null)
-        echo $PCT
+        echo "$PCT%" >&2
```
I intentionally didn't log this to `/tmp/debug` as I didn't think it was useful outside of watching it on the console. I can switch it to the `log` method if you think it has value.

As far as writing to stderr, everything being printed on the serial console is being written to stderr (`log` is doing it as well).

Will update the variable wrapping.

dustymabe commented on coreos-installer:
```diff
@@ -551,58 +556,93 @@ validate_image() {
+Options:
+    -d DEVICE   Install to the given device, alternatively specify coreos.inst.install_dev on the kcmdline.
+    -i IGNITION The URL (or path) to the given Ignition config, alternatively specify coreos.inst.ignition_url on the kcmdline.
+    -b BASEURL  The URL to the image, alternatively specify coreos.inst.image_url on the kcmdline.
```
when viewing this output on a limited VGA console (i.e. virt-manager) the text flows to the next line. and looks jumbled. Maybe we should wrap at 75 characters?

arithx replied:
That's a good point, I'll go full pep8 and wrap at 72 chars.

arithx commented:
Update pushed.

- Changed `Embedding` capitalization
- Image download percent output string format
- Help message character line count restricted to 72 per line.

dustymabe commented:
LGTM
